### DISCUSSION TITLE
cannot build and run prometheus

### DISCUSSION THREAD

**Discussion by sababa976** [prometheus/prometheus#13960] [Q&A]
hello everyone!
i want to start working with prometheus and use it's open source!
i am quite new to this so i need a little bit of help, i have looked and asked many people i know but i didn't get the help i needed so i hope this is the right place for that!
when i use the "make build" command i get this error:
up to date, audited 2756 packages in 8s

183 packages are looking for funding
  run `npm fund` for details

135 vulnerabilities (1 low, 104 moderate, 26 high, 4 critical)

To address issues that do not require attention, run:
  npm audit fix

To address all issues (including breaking changes), run:
  npm audit fix --force

Run `npm audit` for details.
cd web/ui && CI="" npm run build

> prometheus-io@0.51.2 build
> GENERATE_SOURCEMAP=false bash build_ui.sh --all

'GENERATE_SOURCEMAP' is not recognized as an internal or external command,
operable program or batch file.
make: *** [ui-build] Error 1

it confuses me a lot since i dont understand what the problem is, can someone help me? give me an advice?
your help is much appreciated!

**Comment by Epy** [comment]
Hello
As I don't know much about npm etc. I used the Go binary instead. Maybe it could be a solution for you too ?

You just have to make sure your system can execute Go and create/copy a service file (for systemd or equivalent). Ask for more details if you need :)

**Reply by sababa976** [reply]
Thank you for the help!
could you please give a bit more details about what to try?

**Reply by Epy** [reply]
``` bash
# as root or using sudo with an authorized user, it's even better
# a dedicated user without home dir for prometheus to run
useradd -M -U prometheus

# I choose /opt for Go apps
cd /opt/
mkdir prometheus
# check the last version of prometheus and arch of your distribution and adapt this line to those values

wget https://github.com/prometheus/prometheus/releases/download/v2.51.2/prometheus-2.51.2.linux-amd64.tar.gz
# extract in /opt/prometheus directory then go in
tar -C prometheus -xvfz prometheus-2.51.2.linux-amd64.tar.gz && cd prometheus

# Edit the example config, following official documentation for example (monitoring prometheus itself and some linux nodes to test it)
vim prometheus.yml 
# Check it with:
./promtool check config /opt/prometheus/prometheus.yml
# Start it manually to test your config and then go to http://your-server-ip:9090/ 
./prometheus --config.file=prometheus.yml
# Now the full directory and files belongs to the user prometheus
chown -R prometheus:prometheus /opt/prometheus

# Creating the service file for systemd
vim /etc/systemd/system/prometheus.service
systemctl daemon-reload
systemctl status prometheus.service 
systemctl start prometheus.service 
```

The service file is like this:
I think it was in the documentation
```
[Unit]
Description=Prometheus Server
Documentation=https://prometheus.io/docs/introduction/overview/
After=network-online.target

[Service]
User=prometheus
Group=prometheus
Restart=on-failure
ExecStart=/opt/prometheus/prometheus \
  --config.file=/opt/prometheus/prometheus.yml \
  --storage.tsdb.path=/opt/prometheus/data \
  --storage.tsdb.retention.time=30d

[Install]
WantedBy=multi-user.target
```
Note: I'm not an expert, it may contains mistakes or not be the most secure way to do it (for example I didn't check the binary with sha and GPG, which is really recommended)
I'll try to make an up-to-date automated Debian package one day or use Ansible to automate this process like suggested in the documentation
